mne/report.py

```diff
@@ -875,7 +875,173 @@ class Report(object):
+        ...                          'sample_audvis-cov.fif')
+        >>> report = Report(subject='sample',
+        ...                 subjects_dir=subjects_dir,
+        ...                 info_fname=info_fname,
+        ...                 cov_fname=cov_fname,
+        ...                 verbose=True)                 # doctest: +SKIP
+        >>> report.parse_folder(path)                     # doctest: +SKIP
+        >>> report.save()                                 # doctest: +SKIP
+
+    The python interface has greater flexibility compared to the command line
+    interface. Custom plots can be added to the report. Let us first generate a
+    custom plot::
+
+        >>> from mne import read_evokeds
+        >>> fname_evoked = os.path.join(path, 'MEG', 'sample',
+        ...                             'sample_audvis-ave.fif')
+        >>> evoked = read_evokeds(fname_evoked,
+        ...                       condition='Left Auditory',
+        ...                       baseline=(None, 0),
+        ...                       verbose=True)             # doctest: +SKIP
+        >>> fig = evoked.plot(show=False, time_unit='s')    # doctest: +SKIP
+
+    To add the custom plot to the report, do::
+
+        >>> report.add_figs_to_section(fig, captions='Left Auditory',
+                                       section='evoked')    # doctest: +SKIP
+        >>> report.save('report.html', overwrite=True)      # doctest: +SKIP
+
+
+    The MNE report command internally manages the sections so that plots
+    belonging to the same section are rendered consecutively. Within a section,
+    the plots are ordered in the same order that they were added using the
+    :meth:`~mne.Report.add_figs_to_section` command. Each section is identified
+    by a toggle button in the navigation bar of the report which can be used to
+    show or hide the contents of the section. To toggle the show/hide state of
+    all sections in the html report, press :kbd:`t`.
+
+    Saving to HTML is a write-only operation, meaning that we cannot read an
+    ``.html`` file back as a :class:`~mne.Report` object. In order to be able
+    to read it back, we can save it as an HDF5 file::
+
+        >>> from mne import open_report
+        >>> report.save('report.h5', overwrite=True)       # doctest: +SKIP
+        >>> report_from_disk = open_report('report.h5')    # doctest: +SKIP
+
+    This allows multiple scripts to add figures to the same report. To make
+    this even easier, :class:`mne.Report` can be used as a context manager,
+    allowing you to do this::
+
+        >>> with open_report('report.h5') as report:
+        ...    report.add_figs_to_section(fig,
+        ...                               captions='Left Auditory',
+        ...                               section='evoked',
+        ...                               replace=True)      # doctest: +SKIP
+        >>>    report.save('report.html', overwrite=True)    # doctest: +SKIP
```

Comment:
Actually maybe this can just be moved over to:

https://mne.tools/dev/auto_examples/visualization/plot_make_report.html#sphx-glr-auto-examples-visualization-plot-make-report-py

And moved to `tutorials`. Maybe in another PR.